pkg/descheduler/strategies/node_affinity.go

```diff
@@ -49,7 +49,7 @@ func RemovePodsViolatingNodeAffinity(ctx context.Context, client clientset.Inter
 
 				for _, pod := range pods {
 					if pod.Spec.Affinity != nil && pod.Spec.Affinity.NodeAffinity != nil && pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
-						if !nodeutil.PodFitsCurrentNode(pod, node) && nodeutil.PodFitsAnyNode(pod, nodes) {
+						if !nodeutil.PodFitsCurrentNode(pod, node) && (nodeutil.PodFitsAnyNode(pod, nodes) || strategy.Params.NodeSelection.DegradationAllowed) {
```

Comment:
`(strategy.Params.NodeSelection.DegradationAllowed || nodeutil.PodFitsAnyNode(pod, nodes))` so `nodeutil.PodFitsAnyNode` does not have to be called at all when `strategy.Params.NodeSelection.DegradationAllowed` is true. `nodeutil.PodFitsAnyNode` might eventually become expensive to compute.